deps/llvm.mk

```diff
@@ -334,6 +334,36 @@ $(eval $(call bb-install,lld,LLD,false,true))
 $(eval $(call bb-install,clang,CLANG,false,true))
 $(eval $(call bb-install,llvm-tools,LLVM_TOOLS,false,true))
 
+install-llvm-tools:
```

Comment:
This should be written as part of the target that generates the `manifest` file for LLVM, and not something that runs after the install is officially completed

Reply:
I was looking at this with Gabriel, and the rule that generates the manifest is the thing that does the unpacking, so I think we have to run it after the manifest is generated (but before install is completed).  Putting it as part of the `install-` target seems to satisfy those criteria.

Reply:
The manifest is generated after the completion of the install

Reply:
The unpacking of the files is done in the target that generates the `manifest` file: https://github.com/JuliaLang/julia/blob/master/deps/tools/bb-install.mk#L51-L61

So there is no opportunity to add something here that runs after things have been extracted, but before the manifest file exists.

Reply:
Other versions of the install generator have usually taken an argument of custom steps that are needed to run. Take for example MBEDTLS_INSTALL, which has both a custom script for staging and custom post-install steps

Reply:
Yes, we could include custom post-install steps.  Is the reason you don't want to use the `install-*` targets that other downstream targets are depending on the manifest rather than the `install` target?

Reply:
yeah, other targets might depend on that, and if this step fails, it will not be re-run

also that putting stateful commands into phony targets can get pretty hairy

Reply:
So the way forward here is to make `bb-install` take a custom post install script and define a fallback?

Reply:
yeah